chia/wallet/did_wallet/did_wallet.py

```diff
@@ -69,11 +69,12 @@ async def create_new_did_wallet(
             raise ValueError("Cannot require more IDs than are known.")
         self.did_info = DIDInfo(None, backups_ids, num_of_backup_ids_needed, [], None, None, None, None, False)
         info_as_string = json.dumps(self.did_info.to_json_dict())
-        self.wallet_info = await wallet_state_manager.user_store.create_wallet(
+        new_wallet_info = await wallet_state_manager.user_store.create_wallet(
```

Comment:
Why do you prefer this?

Reply:
Because the thing being returned could be None, there are now explicit checks testing for None, and the instance variable that the value is being assigned to is of type WalletInfo. When I loaded up the PR in VSCode, the type hinting was showing the type as any, so that didn't seem especially helpful. Maybe it's just a tooling issue, but I generally prefer having clear typing info available. That's why I prefer the explicit type annotations.

Reply:
The other PR that was referenced would fix (at least part of) the `Any` hinting by fixing the cause of that lack of information.  Explicit is nice, but it can also introduce inaccurately wide hints that cause extra work.  I've seen cases where we hinted like `x: Optional[int] = f()` with `def f() -> int:` resulting in various handling of `None` when there's no `None` to be had.  It also means that changing or correcting hints involves every call site instead of just changing the function definition and then letting the new information propagate.

Anyways, there is interest in the explicitness from more than just you so just sharing opinions here for now.

Related to the existing optionality, a create method returning `None` is a bit odd so I made https://github.com/Chia-Network/chia-blockchain/pull/10957.  The call sites either didn't handle the case or raised an exception anyways.